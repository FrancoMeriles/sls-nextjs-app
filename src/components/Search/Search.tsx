import React from 'react'
import { Button, Input, InputGroup, InputRightElement } from '@chakra-ui/react'
import { SearchIcon } from '@chakra-ui/icons'
import { useRouter } from 'next/router'

const Search = () => {
  const router = useRouter()
  const [value, setValue] = React.useState('')
  const handleChange = (event) => setValue(event.target.value)

  const handleFormSubmit = (e) => {
    e.preventDefault()
    router.push({
      pathname: '/products',
      query: {
        q: value,
        page: 1,
      },
    })
  }

  return (
    <form onSubmit={handleFormSubmit}>
      <InputGroup borderColor="transparent">
        <Input
          value={value}
          onChange={handleChange}
          variant="outline"
          placeholder="Buscar"
          size="lg"
          focusBorderColor="gray.100"
          backgroundColor="white"
        />
        <InputRightElement
          height="100%"
          children={
            <Button type="submit" height="100%" variant="link">
              <SearchIcon color="black.500" />
            </Button>
          }
        />
      </InputGroup>
    </form>
  )
}

export default Search
