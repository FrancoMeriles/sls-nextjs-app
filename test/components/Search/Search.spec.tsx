import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import userEvent from '@testing-library/user-event'

import Search from '@base/components/Search'

const mockOnClose = jest.fn()
const onClose = mockOnClose

const mockPushRouter = jest.fn()

jest.mock('next/router', () => ({
  useRouter: () => ({
    push: mockPushRouter,
  }),
}))

describe('Search', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer.create(<Search onClose={onClose} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data', async () => {
    const stringText = 'Hello!'
    const { container } = render(<Search onClose={onClose} />)
    userEvent.type(container.querySelector('.chakra-input'), stringText)
    userEvent.click(screen.getByRole('button'))
    expect(mockPushRouter).toHaveBeenCalledWith({
      pathname: '/products',
      query: { page: 1, q: stringText },
    })
    expect(mockOnClose).toHaveBeenCalled()
  })
})
