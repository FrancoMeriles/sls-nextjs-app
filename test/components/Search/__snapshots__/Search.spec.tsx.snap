// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render and match snapshot with required props 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="chakra-input__group css-1idadyg"
  >
    <input
      className="chakra-input css-cxmp86"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Buscar"
      value=""
    />
    <div
      className="chakra-input__right-element css-1ctuf01"
    >
      <button
        className="chakra-button css-1a05ek7"
        role="button"
        type="submit"
      >
        <svg
          className="chakra-icon css-1ivt62w"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</form>
`;
